File: leetcode_ly/c++/Tree/MinimumDepthOfBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if(!root) return 0;
        else if(root->left && root->right){
            int left = minDepth(root->left);
            int right = minDepth(root->right);
            return (left < right) ? (1 + left) : (1 + right);
        }
        else{
            int left = minDepth(root->left);
            int right = minDepth(root->right);
            return (left > right) ? (1 + left) : (1 + right);
        }
    }
};

File: leetcode_ly/java/Tree/MinimumDepthOfBinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int minDepth(TreeNode root) {
        if(root == null) return 0;
        if(root.left == null) return 1 + minDepth(root.right);
        else if(root.right == null) return 1 + minDepth(root.left);
        else return 1 + Math.min(minDepth(root.left),minDepth(root.right));
    }
}